scripts/webdriverioTester.js

```diff
@@ -78,50 +86,91 @@ const ns = {
   },
 
   /**
+   * Extracts the config.json file from the tests/e2e directory. If it does not exist, it assumes it on TravisCI, and it attempts
+   * to extract the appropriate environment variables
+   * @returns {Promise} It will either resolve with the configFile or reject with an error
+   */
+  extractConfig () {
+    return new Promise((resolve, reject) => {
+      const configDir = path.join(__dirname, '../../..', process.env['E2E_TESTS_DIR'], 'config.json')
+      let configFile = {}
+      try {
+        configFile = JSON.parse(fs.readFileSync(configDir))
+      } catch (e) {
+        console.log(`Since a config.json file does not exist, we are assuming you are on Travis\n\n`)
+      }
+      _.defaults(configFile, {username: 'travis', token: TOKEN_REVOKED})
+      if (configFile.username === 'travis') {
+        console.log(`Your config.json file must contain a valid username and token.
+        Please visit http://wdio.bp.cyaninc.com to sign up to become an authorized third party developer for Ciena. \n\n`)
+        let repo = process.env['TRAVIS_REPO_SLUG'].split('/')
+        let user = repo[0]
+        let sha = process.env['TRAVIS_COMMIT']
```

Comment:
I bet you could factor out all the username/password finding stuff into its own function and cut down the size of this (large) function